pkg/syncer/fullsync.go

```diff
@@ -272,13 +275,67 @@ func CsiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer, vc s
 	go fullSyncDeleteVolumes(ctx, volToBeDeleted, metadataSyncer, &wg, migrationFeatureStateForFullSync, volManager, vc)
 	wg.Wait()
 
+	// Sync VolumeInfo CRs
+	if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+		volumeInfoCRFullSync(ctx, k8sPVMap, vc)
+	}
+
 	cleanupCnsMaps(k8sPVMap)
 	log.Debugf("FullSync: cnsDeletionMap at end of cycle: %v", cnsDeletionMap)
 	log.Debugf("FullSync: cnsCreationMap at end of cycle: %v", cnsCreationMap)
 	log.Infof("FullSync: end")
 	return nil
 }
 
+// volumeInfoCRFullSync creates VolumeInfo CR if it does not already exist for a volume.
+// It also deletes VolumeInfo CR if its corresponding PV does not exist.
+func volumeInfoCRFullSync(ctx context.Context, k8sPvs map[string]string, vc string) {
```

Comment:
Do you only supply k8spvs for the supplied vCenter server?

Reply:
This is not complete yet. I am yet to complete it and test it. Will ping once it is ready for review.

Reply:
As discussed, we are only considering Pvs from a particular VC.